internal/loader/configuration_client_manager.go

```diff
@@ -0,0 +1,416 @@
+			return nil, err
+		}
+		if manager.id, err = parseConnectionString(connectionString, IdSection); err != nil {
+			return nil, err
+		}
+		if staticClient, err = azappconfig.NewClientFromConnectionString(connectionString, clientOptionWithModuleInfo); err != nil {
+			return nil, err
+		}
+	} else {
+		if manager.Credential, err = CreateTokenCredential(ctx, provider.Spec.Auth, provider.Namespace); err != nil {
+			return nil, err
+		}
+		if staticClient, err = azappconfig.NewClient(*provider.Spec.Endpoint, manager.Credential, clientOptionWithModuleInfo); err != nil {
+			return nil, err
+		}
+		manager.Endpoint = *provider.Spec.Endpoint
+	}
+
+	manager.ValidDomain = getValidDomain(manager.Endpoint)
+	manager.StaticClientWrappers = []*ConfigurationClientWrapper{{
+		Endpoint:       manager.Endpoint,
+		Client:         staticClient,
+		BackOffEndTime: metav1.Time{},
+		FailedAttempts: 0,
+	}}
+
+	return manager, nil
+}
+
+func (manager *ConfigurationClientManager) GetClients() []*ConfigurationClientWrapper {
+	currentTime := metav1.Now()
+	clients := make([]*ConfigurationClientWrapper, 0)
+	for _, clientWrapper := range manager.StaticClientWrappers {
+		if currentTime.After(clientWrapper.BackOffEndTime.Time) {
+			clients = append(clients, clientWrapper)
+		}
+	}
+
+	if !manager.ReplicaDiscoveryEnabled {
+		return clients
+	}
+
+	if currentTime.After(manager.lastFallbackClientAttempt.Time.Add(MinimalClientRefreshInterval)) &&
+		(manager.DynamicClientwrappers == nil ||
+			currentTime.After(manager.lastFallbackClientRefresh.Time.Add(FallbackClientRefreshExpireInterval))) {
+		manager.lastFallbackClientAttempt = currentTime
+		url, _ := url.Parse(manager.Endpoint)
+		_ = manager.DiscoverFallbackClients(url.Host)
+	}
+
+	for _, clientWrapper := range manager.DynamicClientwrappers {
+		if currentTime.After(clientWrapper.BackOffEndTime.Time) {
+			clients = append(clients, clientWrapper)
+		}
+	}
+
+	return clients
+}
+
+func (manager *ConfigurationClientManager) RefreshClients() {
+	currentTime := metav1.Now()
+	if manager.ReplicaDiscoveryEnabled &&
+		currentTime.After(manager.lastFallbackClientAttempt.Time.Add(MinimalClientRefreshInterval)) {
+		manager.lastFallbackClientAttempt = currentTime
+		url, _ := url.Parse(manager.Endpoint)
+		_ = manager.DiscoverFallbackClients(url.Host)
+	}
+}
+
+func (manager *ConfigurationClientManager) DiscoverFallbackClients(host string) error {
+	srvTargetHosts, err := QuerySrvTargetHost(host)
+	if err != nil {
+		return err
+	}
+
+	// Shuffle the list of SRV target hosts
+	for i := range srvTargetHosts {
+		j := rand.Intn(i + 1)
+		srvTargetHosts[i], srvTargetHosts[j] = srvTargetHosts[j], srvTargetHosts[i]
+	}
+
+	newDynamicClients := make([]*ConfigurationClientWrapper, 0)
+	for _, host := range srvTargetHosts {
+		if isValidEndpoint(host, manager.ValidDomain) {
+			targetEndpoint := "https://" + host
+			client, err := manager.newConfigurationClient(targetEndpoint)
+			if err != nil {
```

Comment:
This error should be propagated, but current you ignore all the errors from this method at line 166